### PR TITLE
Extend REPLConsole.installInto() to accept argument options

### PR DESCRIPTION
**Pull request by sh19910711**
This pull request is mainly to change `REPLConsole.installInto()` to accept options via arguments, and it also contains the following changes:

* Use all `element.dataset.*` as the options of console if not set values
* Promote commandHandle as a method
  - Alternatively, provide `REPLConsole#request()` overridable-like method
* Rename `data-initial-prompt` => `data-prompt-label` on the console markup
* Add tests for REPLConsole.installInto()

In this pull request, I want to support the below case:

```javascript
// some sandbox-ed area (e.g., Chrome DevTools)
REPLConsole.installInto("console", {
  request: function(method, url, params, callback) {
    browser.api.sendMessage({
      type: "proxy-like-request",
      method: method,
      url: url,
      params: params
    }, function(xhrLikeObject) {
      callback(xhrLikeObject);
    })
  }
});
```

Thanks.

### CODE REVIEW

File: lib/web_console/templates/console.js.erb [hunk 1]
```diff
@@ -54,11 +54,34 @@ var styleElementId = 'sr02459pvbvrmhco';
...
+REPLConsole.prototype.getUrl = function(path) {
+  if (path) {
+    return this.remotePath + '/' + path;
+  } else {
+    return this.remotePath;
+  }
+};
+
+REPLConsole.prototype.commandHandle = function(line) {
+  var _this = this;
```
Comment (gsamokovarov) [inline lib/web_console/templates/console.js.erb]: Can we start naming those `self`? :-)

File: lib/web_console/templates/console.js.erb [hunk 2]
```diff
@@ -54,11 +54,34 @@ var styleElementId = 'sr02459pvbvrmhco';
...
+  var params = 'input=' + encodeURIComponent(line);
+  this.request('PUT', this.getUrl(), params, function(xhr) {
```
Comment (gsamokovarov) [inline lib/web_console/templates/console.js.erb]: What happens when this request fails? I think we should handle both cases, showing an appropriate message when this is failing.

**Comment by sh19910711** [comment]
Updated (v2, 4bfd8c0):

* Add a callback for failed cases to the `request()` function
* Improve error messages handling in console

--

And here is an example of error messages:

gif

Thanks!

**Comment by sh19910711** [comment]
@gsamokovarov Thanks for merging :)

**Pull request merged**